lesson04.md

```diff
@@ -0,0 +1,459 @@
+  // condition is true
+}
+```
+
+---
+
+### If statements
+
+```js
+if (condition) {
+  // your code here
+}
+```
+
+* `condition` must be a boolean
+* The code in the block is executed only if the boolean is `true`
+* If the boolean is `false`, the code block is _not_ executed
+
+---
+
+### if Statements - Examples
+
+```js
+if (food === 'pizza') {
+  console.log('My favorite!');
+}
+```
+
+```js
+if (money < 1000000) {
+  console.log('You are not rich :p');
+}
+```
+
+```js
+if (canSpeakRussian) { // same as: canSpeakRussian === true
+  console.log('Привет!');
+}
+```
+
+---
+
+### if statements - quiz
+
+What does the code below output to `console.log`?
+
+```js
+let language = "German";
+
+if (language === "German") {
+  console.log("Guten Tag");
+}
+if (language === "French") {
+  console.log("Bonjour");
+}
+```
+
+`"Guten Tag"`
+<!-- .element: class="fragment" -->
+
+---
+
+### if statements - quiz
+
+What does the code below output to `console.log`?
+
+```js
+let language = "Spanish";
+
+if (language === "German") {
+  console.log("Guten Tag");
+}
+if (language === "French") {
+  console.log("Bonjour");
+}
+if (language !== "German" && language !== "French") {
+  console.log("Good day");
+}
+```
+
+`"Good day"`
+<!-- .element: class="fragment" -->
+
+---
+
+### Code Blocks: {}
+
+Code blocks contain the code between `{` and `}`, and should be indented for better readability:
+
+```js
+if (language === "German") {
+  // In VSCode, you can indent code by pressing the "Tab" key,
+  // or right-click and choose "Format Document"
+  console.log("Guten Tag");
+}
+```
+
+---
+
+### Code Blocks: {}
+
+All variables defined in code blocks only exist inside the block (technical term: “block scope”)
+
+```js
+let name = "John Doe";
+
+if (name === "John Doe") {
+  let greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  let greeting = "Hello my dear";
+}
+
+console.log(greeting); // ERROR!!!!
+```
+
+---
+
+### Code Blocks: {}
+
+Correct solution:
+
+```js
+let name = "John Doe";
+
+let greeting;
+if (name === "John Doe") {
+  greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  greeting = "Hello my dear";
+}
+
+console.log(greeting); // "Hi dude"
+```
+
+---
+
+### Quiz
+
+What does it print?
+
+```js
+let food = "broccoli";
+
+if (food === "pizza") {
+    console.log("yum yum");
+}
+```
+
+Solution: Nothing, since the condition in the `if` is `false`
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz
+
+What does it print?
+
+```js
+let age = 42;
+
+if (age >= 18) {
+    console.log("you are allowed to drive a car");
+}
+if (age < 18) {
+    console.log("you are not allowed to drive a car");
+}
+```
+
+Solution: First condition is `true`, so it prints "you are allowed to drive a car"
+<!-- .element: class="fragment" -->
+
+---
+
+# Task
+
+Define a variable `hour` and set it to `17`. Make an `if` statement that prints `good day` if `hour` is less than 18. Now change `hour` to `19` and run the code again. What changed?
```

Comment:
I never remember which verb you prefer to use, but on MDN they actually use "declare" for variables: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

"create" is also used in the page.

For function they use a mix of "define" and "declare"

Reply:
*sigh* ok - I don't mind doing a big search/replace to be consistent with MDN :) As C++ developer, "declare" means something else completely. Since in JS we have no memory management, "define" for me is the better expression. Since every variable declaration is also a definition, I guess we can use both.